src/go/rpk/pkg/cli/cmd/redpanda/admin/selftest/selftest.go

```diff
@@ -0,0 +1,344 @@
+		RequestSize: 512,
+		DurationMs:  1000, // 500ms
+		Parallelism: 1,
+		Type:        netcheckTagIdentifier,
+	}
+	return test
+}
+
+func DefaultTests(durationDisk int, durationNet int) []interface{} {
+	tests := make([]interface{}, 0)
+	for _, t := range DefaultDiskcheckTests(durationDisk) {
+		tests = append(tests, t)
+	}
+	for _, t := range DefaultNetcheckTests(durationNet) {
+		tests = append(tests, t)
+	}
+	return tests
+}
+
+// Assembles certain types of pre-canned tests depending on user input
+// 1. onlyDisk - Only runs the throughput/latency disk benchmarks
+// 2. onlyNetwork - Only runs the network benchmark
+// 3. onlyConn - Runs the network benchmark with short duration - no metrics
+// 4. All false - Runs the default which is the combination of option 1 & 2.
+func AssembleTests(onlyDisk bool, onlyNetwork bool, onlyConn bool, durationDisk int, durationNet int) []interface{} {
+	tests := make([]interface{}, 0)
+	if onlyDisk {
+		for _, t := range DefaultDiskcheckTests(durationDisk) {
+			tests = append(tests, t)
+		}
+	} else if onlyNetwork {
+		for _, t := range DefaultNetcheckTests(durationNet) {
+			tests = append(tests, t)
+		}
+	} else if onlyConn {
+		tests = append(tests, DefaultConnectivityTest())
+	} else {
+		tests = DefaultTests(durationDisk, durationNet)
+	}
+	return tests
+}
+
+func NewCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "self-test",
+		Short: "Start, stop and query redpanda self-test runs through the admin listener",
+		Args:  cobra.ExactArgs(0),
+	}
+	cmd.AddCommand(
+		newStartSelfTest(fs),
+		newStopSelfTest(fs),
+		newQuerySelfTest(fs),
+	)
+	return cmd
+}
+
+func newStartSelfTest(fs afero.Fs) *cobra.Command {
+	var (
+		yes              bool
+		diskDurationMs   int
+		netDurationMs    int
+		onNodes          []int
+		onlyDisk         bool
+		onlyNetwork      bool
+		onlyConnectivity bool
+	)
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Starts a new self test run",
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			// Load config settings
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			// Warn user before continuing, proceed only via explicit signal
+			if !yes {
+				fmt.Print("Invoking the redpanda self tests may put excessive pressure on your system. It attempts to benchmark disk and network hardware to find their max throughputs. Its advised to not start the test if there are already large workloads running on the system. To continue type [Y/y]:")
+				scanner := bufio.NewScanner(os.Stdin)
+				scanner.Scan()
+				input := scanner.Text()
+				if input != "y" && input != "Y" {
+					return
+				}
+			}
+
+			// Sanity check that only one of the 'only' flags were enabled
+			onlyFlags := []bool{onlyDisk, onlyNetwork, onlyConnectivity}
+			if countBools(onlyFlags, true) > 1 {
+				fmt.Printf("Only one of the --only flags may be provided\n")
+				return
+			}
```

Comment:
You can do this with cobra

https://github.com/spf13/cobra/blob/main/user_guide.md#flag-groups

```
# Example from the docs
You can also prevent different flags from being provided together if they represent mutually exclusive options such as specifying an output format as either --json or --yaml but never both:

rootCmd.Flags().BoolVar(&u, "json", false, "Output in JSON")
rootCmd.Flags().BoolVar(&pw, "yaml", false, "Output in YAML")
rootCmd.MarkFlagsMutuallyExclusive("json", "yaml")
```